src/infrastructure/fileUploader.repository.ts

```diff
@@ -0,0 +1,58 @@
+import type NotesApiTransport from './transport/notes-api';
+import type { Note } from '@/domain/entities/Note';
+import mime from 'mime';
+import type FileUploaderRepositoryInterface from '@/domain/fileUploader.repository.interface';
+
+/**
+ * Facade for interacting with files
+ */
+export default class FileUploaderRepository implements FileUploaderRepositoryInterface {
+  /**
+   * Transport instance
+   */
+  private readonly transport: NotesApiTransport;
+
+  /**
+   * Repository constructor
+   * @param notesApiTransport - notes api transport instance
+   */
+  constructor(notesApiTransport: NotesApiTransport) {
+    this.transport = notesApiTransport;
+  }
+
+  /**
+   * Upload file data
+   * @param noteId - note indentifier to make attachment
+   * @param fileData - file binary data
+   * @param fileName - file name
+   */
+  public async upload(noteId: Note['id'], fileData: Blob, fileName?: string): Promise<string> {
+    const form = new FormData();
+
+    /**
+     * Get file mime type and define extension by it
+     */
+    const mimeType = fileData.type;
+    const fileExtension = mime.getExtension(mimeType);
```

Comment:
I think, this package is too big for such a small task. Maybe we can create own simple utility for that?

https://bundlephobia.com/package/mime@4.0.3

Reply:
or use the original extension?

Reply:
created utility